src/main/java/org/lanternpowered/server/data/manipulator/DataManipulatorRegistry.java

```diff
@@ -681,10 +681,12 @@ public static DataManipulatorRegistry get() {
                 c -> c.register(Keys.IS_FLYING, false));
         register(FoodData.class, ImmutableFoodData.class,
                 c -> {
-                    c.register(Keys.FOOD_LEVEL, 20, 0, LanternKeys.MAX_FOOD_LEVEL);
+                    c.register(LanternKeys.MAX_EXHAUSTION, 40.0, 0.0, Double.MAX_VALUE);
+                    c.register(Keys.EXHAUSTION, 0.0, 0.0, LanternKeys.MAX_EXHAUSTION);
                     c.register(LanternKeys.MAX_FOOD_LEVEL, 20, 0, Integer.MAX_VALUE);
-                    c.register(Keys.EXHAUSTION, 0.0, 0.0, 40.0);
-                    c.register(Keys.SATURATION, 0.0, 0.0, 5.0);
+                    c.register(Keys.FOOD_LEVEL, 20, 0, LanternKeys.MAX_FOOD_LEVEL);
+                    c.registerWithSuppliedMax(LanternKeys.MAX_SATURATION, 20.0, 0.0, container -> container.get(Keys.FOOD_LEVEL).get().doubleValue());
+                    c.register(Keys.SATURATION, 5.0, 0.0, LanternKeys.MAX_SATURATION);
```

Comment:
Weren't you going to remove the `MAX_SATURATION` key and use the food level as max? Because now will the max saturation never change, well, it can only become smaller if the food level decreases.